editor/history_dock.cpp

```diff
@@ -54,6 +55,8 @@ void HistoryDock::refresh_history() {
 	action_list->clear();
 	bool include_scene = current_scene_checkbox->is_pressed();
 	bool include_global = global_history_checkbox->is_pressed();
+	EditorSettings::get_singleton()->set_project_metadata("history", "include_scene", include_scene);
+	EditorSettings::get_singleton()->set_project_metadata("history", "include_global", include_global);
```

Comment:
This shouldn't be done on every history refresh. You'll need to add a new method to connect to the `toggled` signal.

Reply:
Wouldn't it be the same functionality because `refresh_history` is connected to `toggled` signal?

Reply:
`refresh_history()` is also called on undoredo action.

Reply:
Oh, okay. I'll update it now, thanks.